homework2/main.py

```diff
@@ -0,0 +1,265 @@
+import requests
+from collections import Counter
+from datetime import datetime, timedelta
+import csv
+import copy
+
+
+class MovieDataGrabber:
+    def __init__(self, api_key, num_pages=1):
+        self.api_key = api_key
+        self.num_pages = num_pages
+        self.url = "https://api.themoviedb.org/3/discover/movie"
+        self.data = None
+        self.genres_data = None
+        self.headers = {
+            "accept": "application/json",
+            "Authorization": f"Bearer {self.api_key}"
+        }
+
+        self.params = {
+            'include_adult': False,
+            'include_video': False,
+            'sort_by': 'popularity.desc'
+        }
+        self.fetch_data()
+        self.fetch_genres_data()
+
+    def fetch_data(self):
+        all_data = []
+        for page_num in range(1, self.num_pages + 1):
+            self.params['page'] = page_num
+            response = requests.get(self.url, params=self.params, headers=self.headers)
+            if response.status_code == 200:
+                data = response.json()
+                all_data.extend(data.get('results', []))
+            else:
+                print(f"Error fetching data from page {page_num}")
+
+        self.data = all_data
+
+    def fetch_genres_data(self):
+        try:
+            response = requests.get("https://api.themoviedb.org/3/genre/movie/list?language=en", headers=self.headers)
+            response.raise_for_status()  # Raise an exception for bad responses (e.g., 404, 500)
+
+            genres_data = response.json()
+            genres_data.get('genres', [])
+            self.genres_data = genres_data
+            # return genres_data.get('genres', [])
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error fetching genres data: {e}")
+            # return []
+
+    def get_all_data(self):
+        return self.data
+
+    def get_genres_data(self):
+        return self.genres_data
+
+    def get_data_by_indexes(self, start_index, end_index, step):
+        return self.data[start_index:end_index:step]
+
+    # def get_most_popular_title(self):  #для одного популярного фильма
+    #     all_data = self.fetch_data()
+    #     return max(all_data, key=lambda x: x['popularity']).get('title', '') if all_data else ""
+    def get_most_popular_title(self, count):  # для нескольких популярных фильмов
```

Comment:
sorry, it's too hard to review this PR.
it's too unclear. Let's make a call to discuss